integrations/access/datadog/client.go

```diff
@@ -0,0 +1,236 @@
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+)
+
+const (
+	datadogMaxConns    = 100
+	datadogHTTPTimeout = 10 * time.Second
+	statusEmitTimeout  = 10 * time.Second
+)
+
+const (
+	IncidentWritePermissions = "incident_write"
+)
+
+// Datadog is a wrapper around resty.Client.
+type Datadog struct {
+	// DatadogConfig specifies datadog client configuration.
+	DatadogConfig
+
+	client *resty.Client
+}
+
+// NewDatadogClient creates a new Datadog client for managing incidents.
+func NewDatadogClient(conf DatadogConfig, webProxyAddr string, statusSink common.StatusSink) (*Datadog, error) {
+	client := resty.NewWithClient(&http.Client{
+		Timeout: datadogHTTPTimeout,
+		Transport: &http.Transport{
+			MaxConnsPerHost:     datadogMaxConns,
+			MaxIdleConnsPerHost: datadogMaxConns,
+		}}).
+		SetBaseURL(conf.APIEndpoint).
+		SetHeader("Accept", "application/json").
+		SetHeader("Content-Type", "application/json").
+		SetHeader("DD-API-KEY", conf.APIKey).
+		SetHeader("DD-APPLICATION-KEY", conf.ApplicationKey).
+		OnBeforeRequest(func(_ *resty.Client, req *resty.Request) error {
+			req.SetError(&ErrorResult{})
+			return nil
+		}).
+		OnAfterResponse(onAfterDatadogResponse(statusSink))
+
+	return &Datadog{
+		DatadogConfig: conf,
+		client:        client,
+	}, nil
+}
+
+func onAfterDatadogResponse(sink common.StatusSink) resty.ResponseMiddleware {
+	return func(_ *resty.Client, resp *resty.Response) error {
+		log := logger.Get(resp.Request.Context())
+
+		if sink != nil {
+			status := common.StatusFromStatusCode(resp.StatusCode())
+			// No usable context in scope, use background with a reasonable timeout
+			ctx, cancel := context.WithTimeout(context.Background(), statusEmitTimeout)
+			defer cancel()
+
+			if err := sink.Emit(ctx, status); err != nil {
+				log.WithError(err).Errorf("Error while emitting Datadog Incident Management plugin status: %v", err)
+			}
+		}
+
+		if resp.IsError() {
+			var details string
+			switch result := resp.Error().(type) {
+			case *ErrorResult:
+				details = fmt.Sprintf("http error code=%v, errors=[%v]", resp.StatusCode(), strings.Join(result.Errors, ", "))
+			default:
+				details = fmt.Sprintf("unknown error result %#v", result)
+			}
+			return trace.Errorf(details)
+		}
+		return nil
+	}
+}
+
+// CheckHealth pings Datadog and ensures required permissions.
+func (d *Datadog) CheckHealth(ctx context.Context) error {
+	var result PermissionsBody
+	_, err := d.client.NewRequest().
+		SetContext(ctx).
+		SetResult(&result).
+		Get("permissions")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, permission := range result.Data {
+		if permission.Attributes.Name == IncidentWritePermissions && permission.Attributes.Restricted {
+			return trace.AccessDenied("missing incident_write permissions")
+		}
```

Comment:
Just a nit, but I think we can break the loop as soon as we find the permission we are looking for.

```suggestion
		if permission.Attributes.Name == IncidentWritePermissions {
			if permission.Attributes.Restricted {
				return trace.AccessDenied("missing incident_write permissions")
			}

			return nil
		}
```